### ISSUE TITLE
<br> added after <p>

### ISSUE THREAD

**Issue by FormerLurker** [showdownjs/showdown#705]
This is probably my own problem, but I've struggled for a while and thought I'd ask the expert.

I'm loading .md files via an ajax request, converting them to HTML with your excellent script, and displaying them in the browser.  Everything works GREAT (and was VERY intuitive to use) except one thing:  ```<br>``` tags are added after every paragraph and heading.  Example:

This text
```
lorem ipsum
### ipsum lorum
lorem ipsum
```
is changed to the following html:
```
<p>lorem ipsum</p><br><h3 id="real-time">ipsum lorum</h3><br><p>lorem ipsum</p><br>
```
And it is displaying sort of like this (notice the wide spacing between lines):
<p>lorem ipsum</p><br><h3 id="real-time">ipsum lorum</h3><br><p>lorem ipsum</p><br>
But I expected it to display like this (expected spacing between lines):<br><br>
<p>lorem ipsum</p><h3 id="real-time">ipsum lorum</h3><p>lorem ipsum</p>

I examined the result of the request that is being fed into the makeHtml function, and there is exactly one newline character (\n) at the end of each line, so I don't think it's being caused by the way I'm loading the text.  I've tried simple line breaks and changing the flavor, but neither had any effect.

Thanks for your time AND this awesome tool!

**Comment by FormerLurker** [comment]
One more thing I just noticed (doesn't that always happen when you submit an issue?).  The offending element is NOT a ```<br>``` but is rather a ```\n``` character that is not removed during parsing.  I missed this since it was converted to a ```<br>``` by jquery apparently, though I have not verified where exactly it's being changed to ```<br>```.  I can just do a find and replace on these in the meanwhile as a workaround.  Perhaps this is a feature request since those would normally be ignored by the browser?

In other words, feel free to close this if you're aware that the ```\n``` characters are left in the output html.  Also, thanks again for your time!
